testsuite/features/step_definitions/system_monitoring_steps.rb

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2024 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### This file contains the definitions for the steps extracting and reporting information from the system
+
+When(/^I report the bootstrap duration for "([^"]*)"$/) do |host|
+  next unless $quality_intelligence_mode
+
+  duration = last_bootstrap_duration(host)
+  $quality_intelligence.push_bootstrap_duration(host, duration)
+end
```

Comment:
If all the duration functions were integrated to `QualityIntelligence` (see ...) class you wouldn't need to invoke that each time and do one liner instead.
```suggestion
When(/^I report the bootstrap duration for "([^"]*)"$/) do |host|
  $quality_intelligence.push_bootstrap_duration(host) if $quality_intelligence_mode
end
```

Reply:
I prefer to follow the Single Responsibility Principle and separate the fact of extracting the duration and the action of pushing the metric into different functions.

As example, just imagine that some day we decide to re-use the QualityIntelligence handler in another project, we would not like too much to have specific code related to SUMA as part of that class.

Also, that might be a matter of taste, but I prefer to don't encapsulate more that necessary the actions we do inside a Cucumber step definition, otherwise is harder to understand all the involved components.

Reply:
My initial assumption was that we rather won't use it in other projects because tech stack is too specific for this particular use case. But I am ok with it how it is now.
Closing